third_party/xla_client/computation_client.cc

```diff
@@ -66,7 +66,9 @@ bool IsLocalDevice(
     return true;
   }
   Device device = ParseDevice(mp_device);
-  return device.ordinal == parsed_device.id &&
+  const int devices_per_worker = sys_util::GetEnvInt("TPU_NUM_DEVICES", 8);
```

Comment:
Assuming "TPU" at this layer is something we should try to avoid.
Since we know the device kind, and we have partial configuration, we can extract the base device ordinal from it:

https://github.com/pytorch/xla/pull/1218